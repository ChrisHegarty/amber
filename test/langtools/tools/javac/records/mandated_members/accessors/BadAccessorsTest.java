/*
 * @test /nodynamiccopyright/
 * @summary check that the compiler doesn't accept incorrect accessors
 * @compile/fail/ref=BadAccessorsTest.out -XDrawDiagnostics BadAccessorsTest.java
 */

import java.util.List;

public class BadAccessorsTest {
    record R1(int i, int j, int k, List<String> ls) {
        // accessors has to be public
        int i() { return i; }
        private int j() { return j; }
        protected int k() { return k; }
        // must match type exactly
        public List ls() { return ls; }
    }
}
