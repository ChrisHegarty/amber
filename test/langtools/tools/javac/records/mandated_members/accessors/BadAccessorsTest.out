BadAccessorsTest.java:12:13: compiler.err.method.must.be.public: i
BadAccessorsTest.java:13:21: compiler.err.method.must.be.public: j
BadAccessorsTest.java:14:23: compiler.err.method.must.be.public: k
BadAccessorsTest.java:16:21: compiler.err.accessor.return.type.doesnt.match
4 errors
